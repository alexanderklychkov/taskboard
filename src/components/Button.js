import React from 'react';
import classNames from 'classnames';

const Button = ({
  children, onClick, className, disabled, active, ...attrs
}) => {

  const classes = classNames(
    'btn',
    className,
    { active }
  );

  return (
    <button
      {...attrs}
      className={classes}
      disabled={disabled}
      onClick={onClick}
    >{children}</button>
  );
};

Button.defaultProps = {
  children: 'Default button',
  onClick: () => {},
  className: '',
  disabled: false,
  active: false
}

export default Button;