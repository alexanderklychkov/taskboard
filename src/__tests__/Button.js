import React from 'react';
import Button from '../components/Button';
import renderer from 'react-test-renderer';

test('Срабатывание функции onClick() при нажатии', () => {
  const component = renderer.create(
    <Button>Добавить задачу</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});