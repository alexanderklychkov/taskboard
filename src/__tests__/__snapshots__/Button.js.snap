// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Срабатывание функции onClick() при нажатии 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
>
  Добавить задачу
</button>
`;

exports[`Срабатывание функции onClick() при нажатии 2`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
>
  Добавить задачу
</button>
`;
